src/routes/blogs/nimbleedge-x-onnxruntime/+page.svx

```diff
@@ -0,0 +1,137 @@
+
+Traditionally, recommendation systems utilized only historical customer behavior (e.g. purchases, views, likes), updated weekly, daily, or every few hours. However, pioneering apps such as [LinkedIn](https://www.linkedin.com/blog/engineering/member-customer-experience/near-real-time-features-for-near-real-time-personalization), [Doordash](https://doordash.engineering/2021/03/04/building-a-declarative-real-time-feature-engineering-framework/), and [Pinterest](https://medium.com/pinterest-engineering/how-pinterest-leverages-realtime-user-actions-in-recommendation-to-boost-homefeed-engagement-volume-165ae2e8cde8), have lately incorporated near-real-time and real-time features in recommendations. This has enabled highly fresh recommendations based on user interactions from just a few minutes, or even seconds ago, enabling massive uptick in key business metrics.
+
+### **Challenges in real-time ML on cloud**
+
+## IMAGE HERE
+
+However, building real-time ML system on cloud involves several major challenges:
+
+- **High Compute Costs:** Prohibitively large cloud costs for storing and processing in-session user interaction data in real-time
+- **Time to market and complexity:** Setting up real-time cloud pipelines for ML is complex, and requires significant time and developer bandwidth
+- **Privacy Risk and Compliance:** Users' in-session clickstream data must be sent to cloud servers for processing, raising privacy concerns
+- **Scaling Challenges:** Handling traffic surges requires either massive reserve cloud capacity, or rapid scaling which is difficult to achieve on cloud
+- **High Latency:** High latency due to time required to make cloud API calls from/ to user mobile devices
+
+### **On-device ML as an alternative**
+
+Running real-time ML for recommendations on users' mobile devices is a viable alternative to using cloud. This eliminates most of the challenges associated with cloud, as described above.
+
+- **Cost:** Significantly lower cloud compute requirements through on-device inference and data processing
+- **Privacy:** In-session user interaction data is processed on device, enhancing privacy
+- **Latency:** Most modern smartphones are highly capable of rapid data processing and inference computations
+- **Scaling:** Effortless scaling, as traffic increase translates to more mobile devices available for computation
+
+However, developing and maintaining on-device real-time ML systems is also highly complex, and requires building systems for
+
+1. on-device real-time event capture and storage
+2. on-device feature processing (like rolling window aggregates)
+3. feature syncing from cloud
+4. ML inference execution suitably balancing latency and resource utilization across diverse mobile devices.
```

Comment:
Blog doesn't show bullet points or numbering for this list which seems inconsistent. Should we convert from a numbered list to dashes?

Reply:
Fixed, thanks :)